libraries/standard/mqtt/src/iot_mqtt_validate.c

```diff
@@ -252,6 +260,302 @@ static bool _validatePublish( bool awsIotMqttMode,
     }
 
 cleanup:
+
+    return status;
+}
+
+/*-----------------------------------------------------------*/
+
+static bool _validateListSize( bool awsIotMqttMode,
+                               const IotMqttSubscription_t * pListStart,
+                               size_t listSize )
+{
+    bool status = true;
+
+    /* Check for empty list. */
+    if( pListStart == NULL )
+    {
+        IotLogError( "Subscription list pointer cannot be NULL." );
+
+        status = false;
+        goto cleanup;
+    }
+
+    if( listSize == 0 )
+    {
+        IotLogError( "Empty subscription list." );
+
+        status = false;
+        goto cleanup;
+    }
+
+    /* AWS IoT supports at most 8 topic filters in a single SUBSCRIBE packet. */
+    if( awsIotMqttMode == true )
+    {
+        if( listSize > AWS_IOT_MQTT_SERVER_MAX_TOPIC_FILTERS_PER_SUBSCRIBE )
+        {
+            IotLogError( "AWS IoT does not support more than %d topic filters per "
+                         "subscription request.",
+                         AWS_IOT_MQTT_SERVER_MAX_TOPIC_FILTERS_PER_SUBSCRIBE );
+
+            status = false;
+            goto cleanup;
+        }
+    }
+
+cleanup:
+
+    return status;
+}
+
+/*-----------------------------------------------------------*/
+
+static bool _validateSubscription( bool awsIotMqttMode,
+                                   IotMqttOperationType_t operation,
+                                   const IotMqttSubscription_t * pSubscription )
+{
+    bool status = true;
+    uint16_t i = 0;
+
```

Comment:
need to validate pSubscription to be non null as well

Reply:
It's not possible for pSubscription to be NULL, since it's the address of an array element.